build.gradle

```
@@ -107,23 +139,55 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String checkForEapForFutureRelease(String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?)-EAP-SNAPSHOT.pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    String foundVersion = matcher.group(1)
```

Same here wrt. naming, this should be
```suggestion
    String foundBuildNumber = matcher.group(1)
```
(we need to clearly distinguish b/w `2022.2`- and `222`-formatted version... or  we **p e r i s h**)

---

so when we have numbers without dots (for example 222 or 221345123) -> is it build number, and when when we have for example 2022.2 or 221.345.123 it is version (in the context of the naming) ?

---

so because I can't quite understand it - when here should be ...buildNumber and in line 183 also should be ...buildNumber, so when when should we have version ?

---

```
221.1234.5 - build number
221 - major part of 221.1234.5 build number
2022.1 - release (aka version)
2022.1.1 - also release (aka version)
2022.1 - major part of both 2022.1 and 2021.1.1 releases
```

---

version is only reserved for 2020.1 etc. ?

---

Just a matter of different representation of the same thing! `2021.1`-style versions are used more (but not only) for marketing purposes, `211`-style build numbers are used more internally by JetBrains